crates/pixi_command_dispatcher/src/build_backend_metadata/mod.rs

```diff
@@ -201,6 +216,38 @@ impl BuildBackendMetadataSpec {
         })
     }
 
+    /// Checks if we should skip the metadata cache for this backend.
+    /// Returns true if:
+    /// 1. There's a System backend override (either for this specific backend or all backends)
+    /// 2. OR the original backend spec is System or Path-based
+    fn should_skip_metadata_cache(
+        backend_spec: &BackendSpec,
+        backend_override: &BackendOverride,
+    ) -> bool {
+        let BackendSpec::JsonRpc(json_rpc_spec) = backend_spec;
+
+        // Check if there's a System backend override for this backend
+        // In-memory overrides are deterministic and can use cached metadata
+        let has_system_override = match backend_override {
+            BackendOverride::System(overridden_backends) => overridden_backends
+                .named_backend_override(&json_rpc_spec.name)
+                .is_some(),
+            BackendOverride::InMemory(_) => false,
+        };
+
+        if has_system_override {
+            return true;
+        }
+
+        // Check if the original backend spec is non-binary (System or Path-based)
+        match &json_rpc_spec.command {
+            CommandSpec::System(_) => true,
+            CommandSpec::EnvironmentSpec(env_spec) => {
+                matches!(env_spec.requirement.1, pixi_spec::PixiSpec::Path(_))
```

Comment:
You should check is_mutable instead of whether its path.

Reply:
`is_mutable` exists only on pinned spec not on spec right? And we have not resolved at this point :)